src/analyzer/protocol/http/HTTP.cc

```diff
@@ -1020,6 +1020,32 @@ void HTTP_Analyzer::Undelivered(uint64_t seq, int len, bool is_orig) {
     }
 }
 
+void HTTP_Analyzer::FlipRoles() {
+    analyzer::tcp::TCP_ApplicationAnalyzer::FlipRoles();
+
+    // If FlipRoles() is invoked after we've upgraded to something,
+    // don't do anything. This shouldn't happen as flipping of TCP
+    // connections currently happens before any data is transferred,
+    // but better safe than sorry.
+    if ( upgraded || pia )
+        return;
```

Comment:
If we have a PIA, could we instead pass on the `FlipRoles()`?

Reply:
Also wondering whether this deserves a weird, too, given that we do one below if we've seen requests/replies?

Reply:
> If we have a PIA, could we instead pass on the FlipRoles()?

We add PIA as child analyzer, so that happens during `analyzer::tcp::TCP_ApplicationAnalyzer::FlipRoles()` already.

> Also wondering whether this deserves a weird, too, given that we do one below if we've seen requests/replies?

Yeah, good call.